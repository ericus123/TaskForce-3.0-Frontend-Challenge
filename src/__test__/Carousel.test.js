/* eslint-disable no-undef */
import React from "react";
import { shallow,configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import CarouselSlider from "../comps/Reusables/carousel";
import toJson from "enzyme-to-json";

configure({adapter: new Adapter()});
it("renders carousel without crashing", () => {
    let data = [
     
        {
            "updated": 1629566946881,
            "cases": 46034110,
            "todayCases": 36826,
            "deaths": 972016,
            "todayDeaths": 933,
            "recovered": 36773591,
            "todayRecovered": 26691,
            "active": 8288503,
            "critical": 28636,
            "casesPerOneMillion": 77453.93,
            "deathsPerOneMillion": 1635.45,
            "tests": 637008449,
            "testsPerOneMillion": 1071788.09,
            "population": 594341787,
            "continent": "North America",
            "activePerOneMillion": 13945.68,
            "recoveredPerOneMillion": 61872.8,
            "criticalPerOneMillion": 48.18,
            "continentInfo": {
                "lat": 31.6768272,
                "long": -146.4707474
            },
            "countries": [
                "Anguilla",
                "Antigua and Barbuda",
                "Aruba",
                "Bahamas",
                "Barbados",
                "Belize",
                "Bermuda",
                "British Virgin Islands",
                "Canada",
                "Caribbean Netherlands",
                "Cayman Islands",
                "Costa Rica",
                "Cuba",
                "Cura√ßao",
                "Dominica",
                "Dominican Republic",
                "El Salvador",
                "Greenland",
                "Grenada",
                "Guadeloupe",
                "Guatemala",
                "Haiti",
                "Honduras",
                "Jamaica",
                "Martinique",
                "Mexico",
                "Montserrat",
                "Nicaragua",
                "Panama",
                "Saint Kitts and Nevis",
                "Saint Lucia",
                "Saint Martin",
                "Saint Pierre Miquelon",
                "Saint Vincent and the Grenadines",
                "Sint Maarten",
                "St. Barth",
                "Trinidad and Tobago",
                "Turks and Caicos Islands",
                "USA"
            ]
        },
    ];
    let wrapper = shallow(<CarouselSlider data={data}/>);
    expect(toJson(wrapper)).toMatchSnapshot();

});

