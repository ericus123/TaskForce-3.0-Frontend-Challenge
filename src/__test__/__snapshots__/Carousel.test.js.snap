// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders carousel without crashing 1`] = `
<Carousel
  additionalTransfrom={0}
  arrows={true}
  autoPlay={true}
  autoPlaySpeed={3000}
  centerMode={false}
  className=""
  containerClass="carousel-container"
  customTransition="all 1"
  dotListClass=""
  draggable={false}
  focusOnSelect={false}
  infinite={true}
  itemClass="carousel-item-padding-40-px"
  keyBoardControl={true}
  minimumTouchDrag={80}
  pauseOnHover={true}
  removeArrowOnDeviceType={
    Array [
      "tablet",
      "mobile",
    ]
  }
  renderArrowsWhenDisabled={false}
  renderButtonGroupOutside={false}
  renderDotsOutside={false}
  responsive={
    Object {
      "desktop": Object {
        "breakpoint": Object {
          "max": 3000,
          "min": 1024,
        },
        "items": 2.6,
        "slidesToSlide": 1.7,
      },
      "mobile": Object {
        "breakpoint": Object {
          "max": 464,
          "min": 0,
        },
        "items": 1,
        "slidesToSlide": 1,
      },
      "tablet": Object {
        "breakpoint": Object {
          "max": 1024,
          "min": 464,
        },
        "items": 2,
        "slidesToSlide": 2,
      },
    }
  }
  showDots={false}
  sliderClass=""
  slidesToSlide={1}
  ssr={true}
  swipeable={false}
  transitionDuration={1000}
>
  <Col
    className="col-continent"
    key="0"
  >
    <Col
      className="col-1-continent"
    >
      <h5
        className="text-center cont-name mt-5 b"
      >
        NORTH AMERICA
      </h5>
      <h2
        className="text-center mt-4 cont-num"
      >
        36,826
      </h2>
      <h6
        className="text-center mt-1"
        style={
          Object {
            "color": "black",
          }
        }
      >
        New Cases
      </h6>
      <p
        className="text-center cases-total mt-5"
      >
        All cases: 
        46,034,110
      </p>
    </Col>
    <Col
      className="col-2-continent"
    >
      <div
        className="col-2-div"
      >
        <h2
          className="text-center mt-1 col-2-count"
        >
          933
        </h2>
        <h6
          className="text-center mt-1 new-tit"
        >
          New deaths
        </h6>
        <p
          className="text-center deaths-total "
        >
          Total deaths: 
          972,016
        </p>
      </div>
      <hr />
      <div
        className="col-2-div"
      >
        <h2
          className="text-center mt-1 col-2-count"
        >
          36,773,591
        </h2>
        <h6
          className="text-center mt-1 new-tit"
        >
          Newly recovered
        </h6>
        <p
          className="text-center deaths-total "
        >
          Total recovered:
          26,691
        </p>
      </div>
      <hr />
      <div
        className="col-2-div"
      >
        <h2
          className="text-center mt-1 col-2-count"
        >
          -
        </h2>
        <h6
          className="text-center mt-1 new-tit"
        >
          New vaccinated
        </h6>
        <p
          className="text-center deaths-total "
        >
          -
        </p>
      </div>
    </Col>
  </Col>
</Carousel>
`;
